rskj-core/src/main/java/co/rsk/core/bc/BlockExecutor.java

```diff
@@ -306,6 +311,10 @@ private BlockResult executeInternal(
 
             executedTransactions.add(tx);
 
+            if (this.registerProgramResults) {
```

Comment:
mmm I think we should stop adding more resposabilities to this executor, we can have a special one for automine and build on top of that

Reply:
Yes, but it would be a greater change. The block executor historically had taken TOO MUCH responsabilities to be refactored, in an improvement task